mldsa/mldsa_native.h

```diff
@@ -422,27 +422,20 @@ int MLD_API_NAMESPACE(open)(
     const uint8_t pk[MLDSA_PUBLICKEYBYTES(MLD_CONFIG_API_PARAMETER_SET)]);
 
 /*************************************************
- * Hash algorithm enumeration for pre-hash functions
+ * Hash algorithm constants for pre-hash functions
  **************************************************/
-#ifndef MLD_PRE_HASH_ENUM
-#define MLD_PRE_HASH_ENUM
-typedef enum
-{
-  MLD_SHA2_224,
-  MLD_SHA2_256,
-  MLD_SHA2_384,
-  MLD_SHA2_512,
-  MLD_SHA2_512_224,
-  MLD_SHA2_512_256,
-  MLD_SHA3_224,
-  MLD_SHA3_256,
-  MLD_SHA3_384,
-  MLD_SHA3_512,
-  MLD_SHAKE_128,
-  MLD_SHAKE_256
-} mld_hash_alg_t;
-
-#endif /* !MLD_PRE_HASH_ENUM */
+#define MLD_PREHASH_SHA2_224 1
+#define MLD_PREHASH_SHA2_256 2
+#define MLD_PREHASH_SHA2_384 3
+#define MLD_PREHASH_SHA2_512 4
+#define MLD_PREHASH_SHA2_512_224 5
+#define MLD_PREHASH_SHA2_512_256 6
+#define MLD_PREHASH_SHA3_224 7
+#define MLD_PREHASH_SHA3_256 8
+#define MLD_PREHASH_SHA3_384 9
+#define MLD_PREHASH_SHA3_512 10
+#define MLD_PREHASH_SHAKE_128 11
+#define MLD_PREHASH_SHAKE_256 12
```

Comment:
This is now re-defined 3 times in a multi-level build, but for syntactically equal `#define`s, that's OK, and the situation is already the same for e.g. the various size constants.